plugins/systems/move/Move.cpp

```diff
@@ -24,26 +25,50 @@ void Systems::MoveSystem::run(Engine::GameEngine &engine)
     try {
         auto &posComponents = reg.componentManager().getComponents<Components::Position>();
         auto &velComponents = reg.componentManager().getComponents<Components::Velocity>();
+        auto &deathRangeComponent = reg.componentManager().getComponents<Components::DeathRange>();
 
         size_t i = 0;
         for (i = 0; i < posComponents.size() && i < velComponents.size(); i++) {
-            auto &pos = posComponents[i];
-            auto &vel = velComponents[i];
+            try {
+                auto &pos = posComponents[i];
+                auto &vel = velComponents[i];
 
-            if (!pos || !vel)
+                pos->x += vel->x;
+                pos->y += vel->y;
+                engine.updateComponent(i, pos->getId(), pos->serialize());
+                if (vel->diminishingFactor == 0)
+                    continue;
+                float factor = (float)vel->diminishingFactor / 100;
+                vel->floatX *= factor;
+                vel->floatY *= factor;
+                vel->x = (int)vel->floatX;
+                vel->y = (int)vel->floatY;
+                engine.updateComponent(i, vel->getId(), vel->serialize());
+            } catch (std::exception &e) {
+                std::cerr << "Error: " << e.what() << std::endl;
                 continue;
-            pos->x += vel->x;
-            pos->y += vel->y;
-            engine.updateComponent(i, pos->getId(), pos->serialize());
-            if (vel->diminishingFactor == 0)
+            }
+        }
+        // std::cerr << "DeathRange size: " << deathRangeComponent.size() << std::endl;
+        // std::cerr << "Position size: " << posComponents.size() << std::endl;
+        for (i = 0; i < posComponents.size() && i < deathRangeComponent.size(); i++) {
+            std::cerr << "Checking death range for entity " << i << std::endl;
+            try {
+                auto &drange = deathRangeComponent[i];
+                auto &pos = posComponents[i];
+
+                if (!pos || !drange)
+                    continue;
+                if (pos->x > drange->maximum || pos->x < drange->minimum) {
+                    reg.killEntity((ECS::Entity)i);
+                    std::cerr << "Entity " << i << " has been killed due to death range" << std::endl;
```

Comment:
debug print ?

Reply:
This one is useful to make sure the right Entity has been killed, as we had issues with swapping Entity indexes